src/topology/config/watcher.rs

```diff
@@ -0,0 +1,131 @@
+use crate::Error;
+use notify::{raw_watcher, Op, RawEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use std::{
+    path::{Path, PathBuf},
+    sync::mpsc::{channel, Receiver},
+    thread,
+    time::Duration,
+};
+
+/// Per notify own documentation, it's advised to have delay of more than 30 sec,
+/// so to avoid receiving repetitions of previous events on macOS.
```

Comment:
So this functionality is only relevant on Mac OS? Should we gate it behind that target?

Reply:
No, that functionality is relevant on all OS-s, it's just that Mac OS is a special case and we took that into consideration before, but not any more as it was introducing too much lag.

Reply:
If we are not considering it anymore should we remove this comment?

Reply:
It's better it being there for documentation, since ,for it to not cause issues, we are depending on other parts of the code base.